### PR TITLE
SNOW-1692063 - More functions for numpy compatibility

### PR DESCRIPTION
**Pull request by sfc-gh-jkew**
- logical comparison functions
- a few more math functions
- modified chain testing test to work with operations that result in mixed types

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,8 +53,9 @@
 - Added support for `DataFrame.tz_localize` and `Series.tz_localize`.
 - Added support for `DataFrame.tz_convert` and `Series.tz_convert`.
 - Added support for applying Snowpark Python functions (e.g., `sin`) in `Series.map`, `Series.apply`, `DataFrame.apply` and `DataFrame.applymap`.
-- Added support for `np.subtract`, `np.multiply`, `np.divide`, `np.true_divide`
-- Added support for tracking usages of `__array_ufunc__`
+- Added support for `np.subtract`, `np.multiply`, `np.divide`, and `np.true_divide`.
```
Comment (sfc-gh-nkumar) [inline CHANGELOG.md:56]: We have released 1.23.0. This will be moved to new release version. Please rebase to get updated changelog.

**Review by sfc-gh-nkumar — APPROVED** [review]
Please move changelog to new release version. Otherwise looks good.

File: CHANGELOG.md [hunk 2]
```diff
@@ -53,8 +54,8 @@
...
```
Comment (sfc-gh-yzou) [inline CHANGELOG.md:57]: we probably want to keep the original word, since it has already be published

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@
 ### Snowpark pandas API Updates
 
 #### New Features
+- Added support for `np.float_power`, `np.mod`, `np.remainder`, `np.greater`, `np.greater_equal`, `np.less`, `np.less_equal`, `np.not_equal`, and `np.equal`.
```
Comment (sfc-gh-yzou) [inline CHANGELOG.md:12]: maybe let's call this "added numpy compatibility support" to make a difference from the regular Snowpark pandas support

File: tests/integ/modin/test_numpy.py [hunk 4]
```diff
@@ -69,9 +69,25 @@ def test_logical_operators():
 
 
 @pytest.mark.parametrize(
-    "np_ufunc", [np.add, np.subtract, np.divide, np.multiply, np.true_divide]
+    "np_ufunc",
+    [
+        np.add,
```
Comment (sfc-gh-yzou) [inline tests/integ/modin/test_numpy.py:74]: For many of those, do we have any test with data that has NULL or NaN inside, i believe they might have some difference there, most about the mapping to Snowpark Pandas, for example, are they mapped to snowpark pandas missing value or actual NaN.
Maybe we can add some of those test in a different pr

**Review by sfc-gh-yzou — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (sfc-gh-jkew) [reply CHANGELOG.md:57]: done

**Pull request merged**